src/it/scala/com/ing/wbaa/gargoyle/sts/StsServiceItTest.scala

```diff
@@ -4,98 +4,116 @@ import akka.actor.ActorSystem
+      override protected[this] def keycloakSettings: GargoyleKeycloakSettings = gargoyleKeycloakSettings
+    }.keycloackToken(formData).map(testCode(_))
+  }
+
   // Fixture for starting and stopping a test proxy that tests can interact with.
-  def withTestStsService(testCode: Authority => Assertion): Future[Assertion] = {
-    val stsProxy = new GargoyleStsService
-      with OAuth2TokenVerifier
+  def withTestStsService(testCode: Authority => Future[Assertion]): Future[Assertion] = {
+    val sts = new GargoyleStsService
+      with KeycloakTokenVerifier
       with TokenService
       with TokenXML
       with UserService {
       override implicit def system: ActorSystem = testSystem
 
       override def httpSettings: GargoyleHttpSettings = gargoyleHttpSettings
+
+      override protected[this] def keycloakSettings: GargoyleKeycloakSettings = gargoyleKeycloakSettings
+    }
+    sts.startup.flatMap { binding =>
+        testCode(Authority(Host(binding.localAddress.getAddress), binding.localAddress.getPort))
+          .andThen{case _ => sts.shutdown()}
     }
-    stsProxy.startup
-      .map { binding =>
-        val authority = Authority(Host(binding.localAddress.getAddress), binding.localAddress.getPort)
-        testCode(authority)
-      }
-      .andThen { case _ => stsProxy.shutdown() }
   }
 
-  def withAwsClient(testCode: AWSSecurityTokenService => Assertion): Future[Assertion] =
+  def withAwsClient(testCode: AWSSecurityTokenService => Future[Assertion]): Future[Assertion] =
     withTestStsService { authority =>
       val stsAwsClient: AWSSecurityTokenService = stsClient(authority)
-
-      try {
-        testCode(stsAwsClient)
-      } finally {
-        stsAwsClient.shutdown()
-      }
+        testCode(stsAwsClient).andThen{case _ => stsAwsClient.shutdown()}
     }
 
   "STS getSessionToken" should {
+
     "return credentials for valid token" in withAwsClient { stsAwsClient =>
-      val credentials = stsAwsClient.getSessionToken(new GetSessionTokenRequest()
-        .withTokenCode("validToken"))
-        .getCredentials
-
-      assert(credentials.getAccessKeyId == "accesskey")
-      assert(credentials.getSecretAccessKey == "secretkey")
-      assert(credentials.getSessionToken == "okSessionToken")
-      assert(credentials.getExpiration.getTime <= (System.currentTimeMillis() + 3600 * 1000))
+      withOAuth2TokenRequest(validCredentials) { keycloakToken =>
+        val credentials = stsAwsClient.getSessionToken(new GetSessionTokenRequest()
+          .withTokenCode(keycloakToken.access_token))
+          .getCredentials
+
+        assert(credentials.getAccessKeyId == "accesskey")
+        assert(credentials.getSecretAccessKey == "secretkey")
+        assert(credentials.getSessionToken == "okSessionToken")
+        assert(credentials.getExpiration.getTime <= (System.currentTimeMillis() + 3600 * 1000))
+      }
     }
 
     "throw AWSSecurityTokenServiceException because invalid token" in withAwsClient { stsAwsClient =>
```

Comment:
I think you can remove these tests with invalid tokens as you've tested that in your new test fro KeyCloakTokenVerifier right?